### ISSUE TITLE
My validation set results and insufficient memory on the test set

### ISSUE THREAD

**Issue by Peterchen111** [TuSimple/simpledet#178]
Hello author, my results in CoCo minival set are one point lower than your results in paper . Why?
![result](https://user-images.githubusercontent.com/48093270/62260115-73caf480-b443-11e9-9754-ea3a8355a579.png)
And when I run the test set, I encountered a problem with ‘’can not allocate memory ‘’，my computer memory  is 32g, do you know how to solve it?

**Comment by RogerChern** [comment]
Could you please provide more info like the config used and the GPU involved?

**Comment by Peterchen111** [comment]
> Could you please provide more info like the config used and the GPU involved?

I trained on two 1080ti, kvstoreParam set the GPU to [0,1], and the learning rate is the learning rate of eight GPUs. and 
![1](https://user-images.githubusercontent.com/48093270/62339749-b3f7a900-b50f-11e9-97f9-a0488ef88fd5.png)
the following are the errors encountered during the test.

**Comment by RogerChern** [comment]
Decrease the split size may lower the memory footprint.
https://github.com/TuSimple/simpledet/blob/5981d07a31827e41a68ca92985416a72a944434f/detection_test.py#L64

And what model are you testing, could you paste the config here?

**Comment by Peterchen111** [comment]
> Decrease the split size may lower the memory footprint.
> https://github.com/TuSimple/simpledet/blob/5981d07a31827e41a68ca92985416a72a944434f/detection_test.py#L64
> 
> And what model are you testing, could you paste the config here?

i test in simpledet/config/tridentnet_r101v1c4_c5_1x.py

**Comment by RogerChern** [comment]
```
 tridentnet_r101v1c4_c5_1x.py
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.410
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.618
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.443
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.225
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.462
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.578
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.341
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.534
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.560
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.346
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.623
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.742
 
 tridentnet_r101v1c4_c5_2x.py
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.423
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.631
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.459
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.233
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.470
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.591
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.346
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.543
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.571
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.362
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.631
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.742
```

The default config gives the above result. Are you changing anything?

**Comment by Peterchen111** [comment]
> ```
>  tridentnet_r101v1c4_c5_1x.py
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.410
>  Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.618
>  Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.443
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.225
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.462
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.578
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.341
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.534
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.560
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.346
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.623
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.742
>  
>  tridentnet_r101v1c4_c5_2x.py
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.423
>  Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.631
>  Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.459
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.233
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.470
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.591
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.346
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.543
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.571
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.362
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.631
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.742
> ```
> 
> The default config gives the above result. Are you changing anything?

I see 。But i didn't change anything else. I simply changed the number of GPUs and the learning rate.

**Comment by RogerChern** [comment]
I see. You shall not change the learning rate as it adapts automatically with the number of GPUs.

https://github.com/TuSimple/simpledet/blob/5981d07a31827e41a68ca92985416a72a944434f/config/tridentnet_r101v1c4_c5_1x.py#L161

**Comment by Peterchen111** [comment]
> I see. You shall not change the learning rate as it adapts automatically with the number of GPUs.
> 
> https://github.com/TuSimple/simpledet/blob/5981d07a31827e41a68ca92985416a72a944434f/config/tridentnet_r101v1c4_c5_1x.py#L161
Emm... Okay, I'll try again. Thanks a lot!
